airavata-services/profile-service/profile-service-server/src/main/java/org/apache/airavata/service/profile/handlers/TenantProfileServiceHandler.java

```diff
@@ -183,12 +185,25 @@ public boolean deleteGateway(AuthzToken authzToken, String gatewayId) throws Ten
     public boolean isGatewayExist(AuthzToken authzToken, String gatewayId) throws TenantProfileServiceException, AuthorizationException, TException {
         try {
             Gateway gateway = tenantProfileRepository.getGateway(gatewayId);
```

Comment:
This won't work as is. getGateway() should query by gatewayId where status is APPROVED.  This way if there is an APPROVED gateway with the gatewayId getGateway() should return it.  That means too that the checks on lines 188-189 are unnecessary.